app/controllers/hq/partners_controller.rb

```diff
@@ -6,6 +6,19 @@ def index
     @partners= Partner.all
   end
 
+  def new
+    @partner = Partner.new
```

Comment:
I am still unsure about `build_partner`. There are two things it tries to do: assign the `@partner` instance variable, if necessary, and build up `@partner`'s attributes, if necessary. In order to accomplish this, we are letting go of the Rails convention `params.require(:partner).permit(...)` because sometimes empty params need to be accepted. In light of this, my original comment on #252 still feels valid.

I imagine that `build_partner` would have more value in two cases - if the same additional action needed to be performed after variable/attributes assignment, and/or if we called `build_partner` from a `before_action` for `new`, `create` & `update`. Are there plans for either?

Regardless, could the method be rewritten as
```ruby
def build_partner(attrs)
  @partner ||= Partner.new
  @partner.attributes = attrs
end

# in new
  @partner = build_partner({})


# in update & create
  @partner = build_partner(partner_params)

def partner_params
  params.require(:partner).permit(...)
end
```
An alternative would be `def build_partner(attrs = {})` but then the calls to it would look different from `new` vs `update` & `create`, which is probably less clear. Either way, we are back to using `params.require`, but it still doesn't address @azangru's concern about the benefit of calling the method instead of simply `Partner.new`.